### PR TITLE
Add connection related methods to the core

### PR DESCRIPTION
**Pull request by marinaserranomontes**
## Pull request checklist

- [ ] All tests pass. Demo project builds and runs.
- [ ] I have resolved any merge conflicts. Confirmation: ____

## What's in this pull request?

>...

### CODE REVIEW

File: OTAcceleratorCore/OTMultiPartyCommunicator.m [hunk 1]
```diff
@@ -270,6 +275,8 @@ - (NSError *)disconnect {
     }
     
     self.isCallEnabled = NO;
+    _internalConnectionCount=0;
```
Comment (southfox) [inline OTAcceleratorCore/OTMultiPartyCommunicator.m]: hi @marinaserranomontes the spaces between the = are missing :)

File: OTAcceleratorCore/OTMultiPartyCommunicator.m [hunk 2]
```diff
@@ -385,10 +394,27 @@ - (void)session:(OTSession *)session streamDestroyed:(OTStream *)stream {
     }
 }
 
+- (void)  session:(OTSession*) session
```
Comment (southfox) [inline OTAcceleratorCore/OTMultiPartyCommunicator.m]: there is a double space between (void)   session, I would look the best if there's no space :)

File: OTAcceleratorCore/OTMultiPartyCommunicator.m [hunk 3]
```diff
@@ -385,10 +394,27 @@ - (void)session:(OTSession *)session streamDestroyed:(OTStream *)stream {
...
+    
+    //check creationtime of the connections
+    [self compareConnectionTimeWithConnection: connection];
+}
+
 - (void)sessionDidDisconnect:(OTSession *)session {
     [self notifyAllWithSignal:OTPublisherDestroyed
                    subscriber:nil
                         error:nil];
+    _internalConnectionCount=0;
```
Comment (southfox) [inline OTAcceleratorCore/OTMultiPartyCommunicator.m]: spaces in =, it should look great if it's:
```
_internalConnectionCount = 0;
```

File: OTAcceleratorCore/OTMultiPartyCommunicator.h [hunk 4]
```diff
@@ -169,4 +169,19 @@ typedef void (^OTMultiPartyCommunicatorBlock)(OTCommunicationSignal signal, OTMu
...
+
+/**
+ *  A boolean value to indicate if the connection is the oldest.
+ */
+@property (readonly, nonatomic) BOOL isFirstConnection;
+
+/**
+ *  A string value to indicate the own connectionId.
+ */
+@property (readonly, nonatomic) NSString *selfConnectionId;
```
Comment (Lucashuang0802) [inline OTAcceleratorCore/OTMultiPartyCommunicator.h]: IMHO, I don't suggest to introduce this as you can easily get this info from `OTAccelerator.connection.connectionId`.

File: OTAcceleratorCore/OTMultiPartyCommunicator.m [hunk 5]
```diff
@@ -385,10 +394,27 @@ - (void)session:(OTSession *)session streamDestroyed:(OTStream *)stream {
...
+    [self compareConnectionTimeWithConnection: connection];
+}
+
+- (void)session:(OTSession*) session
+connectionDestroyed:(OTConnection*) connection {
+    _internalConnectionCount--;
+    
+    //check creationtime of the connections
+    [self compareConnectionTimeWithConnection: connection];
+}
```
Comment (Lucashuang0802) [inline OTAcceleratorCore/OTMultiPartyCommunicator.m]: For calculating the `connectionCount`, we can take advantage of the active streams by the following implementation so `connectionCreated:` and `connectionDestroyed:` are not needed to reduce the code complexity. 
```
func connectionCount() -> Int {
        var streams = [String: Int]()
        for stream: OTStream in AppDelegate.getSharedAcceleratorSession().streams.values {
            if streams[stream.connection.connectionId] != nil {
                streams[stream.connection.connectionId]! += 1
            }
            else {
                streams[stream.connection.connectionId] = 1
            }
        }
        return streams.count
    }
```

File: OTAcceleratorCore/OTMultiPartyCommunicator.h [hunk 6]
```diff
@@ -169,4 +169,19 @@ typedef void (^OTMultiPartyCommunicatorBlock)(OTCommunicationSignal signal, OTMu
  */
 @property (readonly, nonatomic) OTStreamVideoType videoType;
 
+/**
+ *  An integer value to indicate the total connections.
+ */
+@property (readonly, nonatomic) NSUInteger connectionCount;
```
Comment (Lucashuang0802) [inline OTAcceleratorCore/OTMultiPartyCommunicator.h]: 👍

File: OTAcceleratorCore/OTMultiPartyCommunicator.h [hunk 7]
```diff
@@ -169,4 +169,19 @@ typedef void (^OTMultiPartyCommunicatorBlock)(OTCommunicationSignal signal, OTMu
...
+
+/**
+ *  A boolean value to indicate if the connection is the oldest.
+ */
+@property (readonly, nonatomic) BOOL isFirstConnection;
```
Comment (Lucashuang0802) [inline OTAcceleratorCore/OTMultiPartyCommunicator.h]: IMHO, let's not introduce this as there is no clean evidence that this is used very often. It's not like `connectionCount`, there are a lot of customers ruquesting that.

**Comment by Lucashuang0802** [comment]
@marcioaffonso IMHO, the `connectionCount` is only the interested API to add. However, we could later add it if there are any requests. I believe we have the implementation on the `wrapper` branchso so we can easily provide if there are any cases for it in the future. Thanks for the great work! 😃

File: OTAcceleratorCore/OTMultiPartyCommunicator.m (hunk 5 above)
Comment (marinaserranomontes) [inline OTAcceleratorCore/OTMultiPartyCommunicator.m]: just to clarify, a new stream doesn't mean a new connection. Imagine I am connected to the session but I am not publishing --> a new connection will be created but not a new stream

Comment (Lucashuang0802) [inline OTAcceleratorCore/OTMultiPartyCommunicator.m]: That's right. Ignore my comment. If this data is provided from the sdk level, that would be better.

File: OTAcceleratorCore/OTMultiPartyCommunicator.m [hunk 8]
```diff
@@ -133,6 +133,9 @@ + (instancetype)sharedInstance {
 @interface OTMultiPartyCommunicator() <OTSessionDelegate, OTSubscriberKitDelegate, OTPublisherDelegate, OTVideoViewProtocol>
 @property (nonatomic) BOOL isCallEnabled;
 @property (nonatomic) NSString *name;
+@property (nonatomic) NSUInteger internalConnectionCount;
```
Comment (Lucashuang0802) [inline OTAcceleratorCore/OTMultiPartyCommunicator.m]: This line should be `@property (nonatomic) NSUInteger connectionCount;` and compiler will match it. there is no need to use `internalConnectionCount`.

File: OTAcceleratorCore/OTMultiPartyCommunicator.m [hunk 9]
```diff
@@ -164,6 +167,8 @@ - (instancetype)initWithName:(NSString *)name {
     
     if (self = [super init]) {
         _name = name;
+        _internalConnectionCount = 0;
+        _connectionsOlderThanMe = 0;
```
Comment (Lucashuang0802) [inline OTAcceleratorCore/OTMultiPartyCommunicator.m]: default value will be `0` for `NSUInteger`, so this assignment is not necessary.

File: OTAcceleratorCore/OTMultiPartyCommunicator.m [hunk 10]
```diff
@@ -605,4 +631,37 @@ - (void)updateSubscriber {
     }
 }
 
+- (NSUInteger)connectionCount {
+    return _internalConnectionCount;
+}
```
Comment (Lucashuang0802) [inline OTAcceleratorCore/OTMultiPartyCommunicator.m]: We can remove this one replacing with the `connectionCount`

File: OTAcceleratorCore/OTMultiPartyCommunicator.h (hunk 7 above)
Comment (marinaserranomontes) [inline OTAcceleratorCore/OTMultiPartyCommunicator.h]: We had some requests about that in Android, so IMHO, it could be interesting to have in iOS as well

**Pull request merged**